### DISCUSSION TITLE
Set ExpiresOnReboot

### DISCUSSION THREAD

**Discussion by TelaKeppi** [Windos/BurntToast#161] [Q&A]
When using Get-BTHistory to get history of previous toasts, there's parameter ExpiresOnReboot, which is always false. Is there a way to set it?

I'm using toasts to notify users to reboot their computers and the notifications always stays after, which is annoying.

**Comment by Windos** [comment]
Strictly speaking... anything is possible, unfortunately for this you've got to drop down to the .NET sort of thing to set it as I haven't exposed it through BurntToast's functions.

First you need to end up with a "Toast" object:

```powershell
$XmlString = @"
<toast>
  <visual>
    <binding template="ToastGeneric">
      <text>Default Notification</text>
      <image src="C:\Images\BurntToast.png" placement="appLogoOverride" hint-crop="circle" />
    </binding>
  </visual>
  <audio src="ms-winsoundevent:Notification.Default" />
</toast>
"@

$AppId = '{1AC14E77-02E7-4E5D-B744-2EB1AE5198B7}\WindowsPowerShell\v1.0\powershell.exe'

$null = [Windows.UI.Notifications.ToastNotificationManager, Windows.UI.Notifications, ContentType = WindowsRuntime]
$null = [Windows.Data.Xml.Dom.XmlDocument, Windows.Data.Xml.Dom.XmlDocument, ContentType = WindowsRuntime]

$ToastXml = [Windows.Data.Xml.Dom.XmlDocument]::new()

$ToastXml.LoadXml($XmlString)

$Toast = [Windows.UI.Notifications.ToastNotification]::new($ToastXml)
```

Then set true on the `ExpiresOnReboot` property:

```powershell
$Toast.ExpiresOnReboot = $true
```

Finally, show the toast:

```powershell
[Windows.UI.Notifications.ToastNotificationManager]::CreateToastNotifier($AppId).Show($Toast)
```

I've got a blog post on how to get the XML out of BurntToast for a notification here: https://toastit.dev/2018/08/21/crouton-8-i-like-your-code-but-i-dont-want-it-on-all-my-computers/

I'm currently working on a big update to BurntToast, I will make sure this is easily settable going forward.

**Reply by Simpuhl** [reply]
I tried this and the notifications are still there during a reboot.
```
    $Load = [Windows.UI.Notifications.ToastNotificationManager, Windows.UI.Notifications, ContentType = WindowsRuntime]
    $Load = [Windows.Data.Xml.Dom.XmlDocument, Windows.Data.Xml.Dom.XmlDocument, ContentType = WindowsRuntime]
    # Load the notification into the required format
    $ToastXML = New-Object -TypeName Windows.Data.Xml.Dom.XmlDocument
    $ToastXML.LoadXml($Toast.OuterXml)

    $Toast = [Windows.UI.Notifications.ToastNotification]::new($ToastXml)

    $Toast.ExpiresOnReboot = $true
        
    # Display the toast notification
    try {
        [Windows.UI.Notifications.ToastNotificationManager]::CreateToastNotifier($App).Show($ToastXml)
    }
```




What do you think?

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
